torch/csrc/api/src/data/init.cpp

```diff
@@ -0,0 +1,479 @@
+#include <torch/csrc/api/include/torch/data/init.h>
+#include <torch/csrc/utils/auto_gil.h>
+#include <torch/csrc/utils/pybind.h>
+#include <torch/data/datasets/chunk.h>
+#include <test/cpp/api/dataloader.h>
+
+#include <c10/macros/Export.h>
+#include <caffe2/serialize/inline_container.h>
+
+#include <ATen/core/function_schema.h>
+
+#include <pybind11/functional.h>
+#include <pybind11/iostream.h>
+
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <utility>
+
+namespace torch {
+namespace data {
+
+using namespace torch::data;
+using namespace torch::data::datasets;
+using namespace torch::data::samplers;
+using namespace torch::serialize;
+
+void init_dataset_bindings(PyObject* module) {
+  init_dataset_bindings_impl(module);
+  init_dataset_bindings_test(module);
```

Comment:
Shall we move init_dataset_binding_ to test directory? I think we should avoid init test only binding in product code.

Reply:
`init_dataset_bindings_test` doesnt really implement any test - bad naming alert. It creates a python binding that can be used  as an example for future implementations. Adding to this document-by-example scenario approach, the bindings are also used used by a Python tests (test_dataloader.py) to illustrate how a Python script would use the new Python API based on bindings for the existing C++ code. The actual python test is just a tiny bonus that came with the package :)

Maybe renaming `init_dataset_bindings_test` to `init_dataset_bindings_example` makes this separation between core code and example clearer. What do you think? I will add comments on the new revision

It is not possible to do bindings on main torch module from test folder without